neonvm/controllers/vm_controller.go

```diff
@@ -800,6 +800,11 @@ func (r *VMReconciler) doReconcile(ctx context.Context, vm *vmv1.VirtualMachine)
 		// do nothing
 	}
 
+	if vm.Status.Phase == vmv1.VmRunning && vm.Spec.TargetRevision != nil {
+		rev := vm.Spec.TargetRevision.WithTime(time.Now())
+		vm.Status.CurrentRevision = &rev
+	}
```

Comment:
I'm also a little nervous here about adding new stuff at the end of `doReconcile`. It's easy to write an early return above, assuming that there's nothing relevant at the bottom of the function.

Have you checked we have no such early returns?

If so: I think it's ok for now, but should be reconsidered as part of broader neonvm-controller refactoring.